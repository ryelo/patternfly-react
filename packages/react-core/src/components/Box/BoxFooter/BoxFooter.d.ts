import { HTMLProps } from 'react';

export interface BoxProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
}

declare const BoxFooter: React.SFC<BoxProps>;

export default BoxFooter;
