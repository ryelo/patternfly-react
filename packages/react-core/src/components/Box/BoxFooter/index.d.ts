export {
  default as BoxFooter,
  BoxFooterProps
} from './BoxFooter';
