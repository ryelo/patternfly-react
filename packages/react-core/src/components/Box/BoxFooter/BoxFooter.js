import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxFooter */
  children: PropTypes.node,
  /** additional classes added to the boxFooter */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxFooter = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxFooter, className)}>
    {children}
  </div>
);

BoxFooter.propTypes = propTypes;
BoxFooter.defaultProps = defaultProps;

export default BoxFooter;
