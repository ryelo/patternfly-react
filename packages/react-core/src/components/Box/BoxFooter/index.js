export { default as BoxFooter } from './BoxFooter';
