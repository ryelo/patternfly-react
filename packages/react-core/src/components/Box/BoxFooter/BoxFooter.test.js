import BoxFooter from './BoxFooter';
import React from 'react';
import { shallow } from 'enzyme';

test('Box has Footer', () => {
  const view = shallow(<BoxFooter>Box Footer</BoxFooter>);
  expect(view).toMatchSnapshot();
});
