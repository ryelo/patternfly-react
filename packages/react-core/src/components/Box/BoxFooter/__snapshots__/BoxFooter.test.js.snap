// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box has Footer 1`] = `
.pf-c-box__footer {
  display: block;
  padding: 2rem 2rem 2rem 2rem;
}

<div
  className="pf-c-box__footer"
>
  Box Footer
</div>
`;
